test/unit/put_signal.c

```diff
@@ -0,0 +1,88 @@
+/*
+ *  Copyright (c) 2018 Intel Corporation. All rights reserved.
+ *  This software is available to you under the BSD license below:
+ *
+ *      Redistribution and use in source and binary forms, with or
+ *      without modification, are permitted provided that the following
+ *      conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/* 
+ * Validate shmem_put_signal operation through blocking API
+ * It mimics the broadcast operation through a ring-based algorithm 
+ * using shmem_put_signal.
+*/
+
+#include <stdio.h>
+#include <shmem.h>
+#include <string.h>
+
+#ifdef ENABLE_SHMEMX_TESTS
```

Comment:
Just a thought:  I think we can avoid this check for `ENABLE_SHMEMX_TESTS` by placing these tests in the `shmemx` directory instead of the `unit` directory.  I'm not actually suggesting this though...  On a related note, as of now I'm putting the wait/test any, all, and some routines in the `shmemx` directory, but I can change that if we want to be consistent.  The shmemx nbi atomics are in `unit` right now...